src/Symfony/Component/Lock/Store/RetryTillSaveStore.php

```diff
@@ -99,4 +104,20 @@ public function exists(Key $key)
     {
         return $this->decorated->exists($key);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsPutOffExpiration(): bool
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsWaitAndSave(): bool
+    {
+        return true;
```

Comment:
`$this->decorated instanceof BlockingStoreInterface && $this->decorated->supportsWaitAndSave()`